bin/tx-prover/src/proxy/mod.rs

```diff
@@ -239,6 +223,27 @@ impl LoadBalancer {
         let workers_count = self.num_workers().await;
         Some(create_workers_updated_response(session, workers_count).await)
     }
+
+    /// Check the health of the workers and returns a list of healthy workers.
+    ///
+    /// Performs a health check on each worker using the gRPC health check protocol. If a worker
+    /// is not healthy, it won't be included in the list of healthy workers.
+    async fn check_workers_health(
+        &self,
+        workers: impl Iterator<Item = &mut Worker>,
+    ) -> Vec<Worker> {
+        let mut healthy_workers = Vec::new();
+
+        for worker in workers {
+            if worker.is_healthy().await {
```

Comment:
Not for this PR but I think it would be nice to parallelize these checks somehow. Since they are sequential, if you have a couple of them that are failing, they can both stall all the checks by the defined timeout amount.

Reply:
Maybe it will be good to add this to the #1009 issue (the one of the retries), and change a bit the scope of that issue as a more general "improve health check" in the proxy side. I'm updating the issue, let me know your opinion.